src/list.go

```diff
@@ -164,11 +165,61 @@ func ListGenericBackends() ([]string, error) {
 
 	var backends []string
 
+	version, err := GetVaultVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	// With the 0.8.3 release of vault the "generic" backend was renamed to "kv". For backwards
+	// compatibility choose the right one based on the vault version.
+	var backendType string
+	if (version.Minor == 8 && version.Patch == 3) || version.Minor >= 9 || version.Major > 1 {
```

Comment:
there is no need to split version into major, minor and patch - you can actually compare strings as well.

Example:
"0.8.3" >= "0.8.3" would return true
"0.8.2" >= "0.8.3" would return false